testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestLdapTrinoJdbc.java

```diff
@@ -102,27 +100,22 @@ public void shouldFailQueryForLdapWithoutPassword()
     @Test(groups = {LDAP, TRINO_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
     public void shouldFailQueryForLdapWithoutSsl()
     {
-        try {
-            DriverManager.getConnection("jdbc:trino://" + prestoServer(), ldapUserName, ldapUserPassword);
-            fail();
-        }
-        catch (SQLException exception) {
-            assertEquals(exception.getMessage(), "Authentication using username/password requires SSL to be enabled");
-        }
+        assertThatThrownBy(() -> DriverManager.getConnection("jdbc:trino://" + prestoServer(), ldapUserName, ldapUserPassword))
+                .isInstanceOf(SQLException.class)
+                .hasMessage("Authentication using username/password requires SSL to be enabled");
     }
 
     @Test(groups = {LDAP, TRINO_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
     public void shouldFailForIncorrectTrustStore()
+            throws Exception
     {
-        try {
-            String url = format("jdbc:trino://%s?SSL=true&SSLTrustStorePath=%s&SSLTrustStorePassword=%s", prestoServer(), ldapTruststorePath, "wrong_password");
-            Connection connection = DriverManager.getConnection(url, ldapUserName, ldapUserPassword);
-            Statement statement = connection.createStatement();
-            statement.executeQuery(NATION_SELECT_ALL_QUERY);
-            fail();
-        }
-        catch (SQLException exception) {
-            assertEquals(exception.getMessage(), "Error setting up SSL: Keystore was tampered with, or password was incorrect");
+        String url = format("jdbc:trino://%s?SSL=true&SSLTrustStorePath=%s&SSLTrustStorePassword=%s", prestoServer(), ldapTruststorePath, "wrong_password");
+        try (Connection connection = DriverManager.getConnection(url, ldapUserName, ldapUserPassword);
+                Statement statement = connection.createStatement()) {
+            // The keystore is not loaded until query is executed
+            assertThatThrownBy(() -> statement.executeQuery(NATION_SELECT_ALL_QUERY))
```

Comment:
test fails because the error seems to happen during `getConnection`. good demo of why we need this refactoring :)

```
java.sql.SQLException: Error setting up SSL: Keystore was tampered with, or password was incorrect
2021-01-09T22:39:27.4653809Z tests               | 	at io.trino.jdbc.TrinoDriverUri.setupClient(TrinoDriverUri.java:283)
2021-01-09T22:39:27.4655239Z tests               | 	at io.trino.jdbc.TrinoDriver.connect(TrinoDriver.java:84)
2021-01-09T22:39:27.4656612Z tests               | 	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
2021-01-09T22:39:27.4658089Z tests               | 	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
2021-01-09T22:39:27.4660460Z tests               | 	at io.trino.tests.jdbc.TestLdapTrinoJdbc.shouldFailForIncorrectTrustStore(TestLdapTrinoJdbc.java:113)
```

Reply:
Thanks for investigating the failure!
as you may have guessed, i didn't run the product tests locally.